### PR TITLE
Adds docstring to event_status_type enum

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Adding docstring.

### CODE REVIEW

File: dpctl/enum_types.py [hunk 1]
```diff
@@ -75,6 +75,19 @@ class backend_type(Enum):
 
 
 class event_status_type(Enum):
+    """
+    An enumeration of SYCL event states.
+
+    ==================   ============
```
Comment (diptorupd) [inline dpctl/enum_types.py:81]: Is there a way to automate generating this table? Do users care about the value of an enum?

Comment (diptorupd) [reply dpctl/enum_types.py:81]: My worry is next time we update the enum, the table is not going to be updated and will be incorrect. It has happened to backend and device type.

Comment (oleksandr-pavlyk) [reply dpctl/enum_types.py:81]: I think it is fine. The `enum` values has to be added to Python class `event_status_type` anyway, and docstring should be updated while extending the enum.

Comment (diptorupd) [reply dpctl/enum_types.py:81]: Ok. Can you fix the docs for the other two enums as well in that case 😄

Comment (oleksandr-pavlyk) [reply dpctl/enum_types.py]: Done. Removed the table, used that space to give an example.

**Pull request merged**